src/config/reader.rs

```diff
@@ -62,6 +62,74 @@ impl ConfigReader {
         Ok(content)
     }
 
+    /// Reads the links in a Config and fill the content
+    #[async_recursion::async_recursion]
+    async fn ext_links(
+        &self,
+        mut config_set: ConfigSet,
+        path: Option<String>,
+    ) -> anyhow::Result<ConfigSet> {
+        let links = config_set.config.links.clone();
+
+        if links.is_empty() {
+            return Ok(config_set);
+        }
+
+        if links.iter().any(|link| link.src.is_empty()) {
+            return Err(anyhow::anyhow!("Link src cannot be empty"));
+        }
+
+        for config_link in links.iter() {
+            let source = if let Ok(data) = self.read_file(&config_link.src).await {
+                data
+            } else {
+                let path = path.clone().unwrap_or_default();
+                let path = PathBuf::from(&path)
+                    .parent()
+                    .unwrap_or(Path::new(""))
+                    .join(&config_link.src);
+
+                self.read_file(path.to_string_lossy()).await?
```

Comment:
Why are we reading the file twice here?

Reply:
if the first read (using absolute path) fails, we go for relative path

Reply:
We shouldn't do that. This should ideally work like how node imports modules. Essentially, if the absolute path exists, use that or if it's a relative path, create an equivalent absolute URL based on the current file path.